src/main/cpp/src/uuid.cu

```diff
@@ -0,0 +1,184 @@
+    uuid_ptr += 2;
+  }
+}
+
+template <int block_size>
+__launch_bounds__(block_size) CUDF_KERNEL void generate_uuids_kernel(
+  cudf::size_type const row_count, char* uuid_chars, curandState* state, int const num_states)
+{
+  auto const start_idx = cudf::detail::grid_1d::global_thread_id();
+  auto const stride    = cudf::detail::grid_1d::grid_stride();
+
+  assert(start_idx < num_states);
+
+  curandState localState = state[start_idx];
+
+  for (cudf::thread_index_type row_idx = start_idx; row_idx < row_count; row_idx += stride) {
+    auto const idx = static_cast<cudf::size_type>(row_idx);
+
+    // curand gets 32-bits random number, cast to 64-bits
+    uint64_t i1 = static_cast<uint64_t>(curand(&localState));
+    uint64_t i2 = static_cast<uint64_t>(curand(&localState));
+    uint64_t i3 = static_cast<uint64_t>(curand(&localState));
+    uint64_t i4 = static_cast<uint64_t>(curand(&localState));
+
+    uint64_t const most  = i1 << 32 | i2;
+    uint64_t const least = i3 << 32 | i4;
+
+    // set the version bits to 4 (UUID version 4): Truly Random or Pseudo-Random
+    auto most_sig_bits = (most & 0xFFFFFFFFFFFF0FFFL) | 0x0000000000004000L;
+
+    // set the variant bits to 2
+    auto least_sig_bits = (least | 0x8000000000000000L) & 0xBFFFFFFFFFFFFFFFL;
+
+    // convert the most and least significant bits to a string format
+    char* uuid_ptr = uuid_chars + idx * CHAR_COUNT_PER_UUID;
+    convert_uuid_to_chars(most_sig_bits, least_sig_bits, uuid_ptr);
+  }
+
+  state[start_idx] = localState;
+}
+
+std::unique_ptr<cudf::column> generate_uuids(cudf::size_type row_count,
+                                             long seed,
+                                             rmm::cuda_stream_view stream,
+                                             rmm::device_async_resource_ref mr)
+
+{
+  // Check if row_count is positive and does not exceed the maximum limit for a column.
+  CUDF_EXPECTS(row_count > 0, "Row count must be positive.");
+  CUDF_EXPECTS(row_count <= std::numeric_limits<cudf::size_type>::max() / CHAR_COUNT_PER_UUID,
+               "Row count exceeds the maximum limit for UUID generation.");
+
+  constexpr int block_size = 128;
+  auto const num_sms       = cudf::detail::num_multiprocessors();
+  int num_blocks_per_sm    = -1;
+
+  // Calculate the maximum number of blocks per multiprocessor
+  CUDF_CUDA_TRY(cudaOccupancyMaxActiveBlocksPerMultiprocessor(
+    &num_blocks_per_sm, generate_uuids_kernel<block_size>, block_size, 0));
+  auto num_states = num_sms * num_blocks_per_sm * block_size;
+
+  // Ensure num_states does not exceed row_count
+  if (num_states > row_count) { num_states = cudf::util::round_up_unsafe(row_count, block_size); }
+
+  // initialize curand states
+  rmm::device_uvector<curandState> states(
+    num_states, stream, cudf::get_current_device_resource_ref());
+  init_curand_kernel<block_size>
+    <<<num_states / block_size, block_size, 0, stream.value()>>>(states.data(), num_states, seed);
```

Comment:
For launching a kernel, use `grid_1d` class to simplify grid calculation and avoid mistake.